src/main/scala/beam/sim/BeamWarmStart.scala

```diff
@@ -2,98 +2,78 @@ package beam.sim
 
 import java.io.File
 import java.nio.file.{Files, Paths}
+import java.util.concurrent.TimeUnit
 
 import akka.actor.ActorRef
 import beam.router.BeamRouter.{UpdateTravelTimeLocal, UpdateTravelTimeRemote}
 import beam.router.LinkTravelTimeContainer
-import beam.sim.config.BeamConfig
+import beam.sim.config.{BeamConfig, BeamExecutionConfig}
 import beam.utils.FileUtils.downloadFile
 import beam.utils.TravelTimeCalculatorHelper
 import beam.utils.UnzipUtility._
 import com.typesafe.scalalogging.LazyLogging
 import org.apache.commons.io.FileUtils.getTempDirectoryPath
 import org.apache.commons.io.FilenameUtils.{getBaseName, getExtension, getName}
 import org.matsim.api.core.v01.Scenario
-import org.matsim.core.config.Config
+import org.matsim.core.config.groups.TravelTimeCalculatorConfigGroup
 import org.matsim.core.router.util.TravelTime
 
 import scala.compat.java8.StreamConverters._
 
 class BeamWarmStart private (beamConfig: BeamConfig, maxHour: Int) extends LazyLogging {
+  val isWarmMode = beamConfig.beam.warmStart.enabled
+  if (!isWarmMode) { // TODO: it will be inlined
+    throw new IllegalStateException("BeamWarmStart cannot be initialized since warmstart is disabled")
+  }
 
   val srcPath = beamConfig.beam.warmStart.path
 
-  /**
-    * check whether warmStart mode is enabled.
-    *
-    * @return true if warm start enabled, otherwise false.
-    */
-  val isWarmMode: Boolean = beamConfig.beam.warmStart.enabled
-
-  /**
-    * initialize travel times.
-    */
   def warmStartTravelTime(beamRouter: ActorRef, scenario: Scenario): Unit = {
-    if (isWarmMode) {
-      read.foreach { travelTime =>
-        beamRouter ! UpdateTravelTimeLocal(travelTime)
-        BeamWarmStart.updateRemoteRouter(scenario, travelTime, maxHour, beamRouter)
-        logger.info("Travel times successfully warm started from")
-      }
+    readTravelTime.foreach { travelTime =>
+      beamRouter ! UpdateTravelTimeLocal(travelTime)
+      BeamWarmStart.updateRemoteRouter(scenario, travelTime, maxHour, beamRouter)
+      logger.info("Travel times successfully warm started from")
     }
   }
 
-  def read: Option[TravelTime] = {
-    if (isWarmMode) {
-      getWarmStartFilePath("linkstats.csv.gz", rootFirst = false) match {
-        case Some(statsPath) =>
-          if (Files.exists(Paths.get(statsPath))) {
-            val travelTime = getTravelTime(statsPath)
-            logger.info("Read travel times from {}.", statsPath)
-            Some(travelTime)
-          } else {
-            logger.warn("Travel times failed to warm start, stats not found at path ( {} )", statsPath)
-            None
-          }
-        case None =>
-          logger.warn("Travel times failed to warm start, stats not found at path ( {} )", srcPath)
+  def readTravelTime: Option[TravelTime] = {
+    getWarmStartFilePath("linkstats.csv.gz", rootFirst = false) match {
+      case Some(statsPath) =>
+        if (Files.isRegularFile(Paths.get(statsPath))) {
```

Comment:
Why this has been changed to `Files.isRegularFile`? What is the reason behind it?

Reply:
`exists` method should be avoided. if there is a directory with that name returns true and you get an exception further. what the method actually needs is a regular file instead of something with that name